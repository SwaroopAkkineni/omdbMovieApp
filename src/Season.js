import React, {Component} from 'react';
import {Dimensions, FlatList, Platform, Text, TouchableOpacity, View} from 'react-native';
import Episode from './Episode';
import styles from '../StyleSheet';

export default class Season extends React.Component {
  state = {
    episodes: [],
	};

  componentDidMount() {
    this.getEpisodes();
  }

  getEpisodes = () => {
    const { navigation } = this.props;
    const showName = navigation.state.params.showName;
    const season = navigation.state.params.season;
    let fetchString = 'http://www.omdbapi.com/?t=';

    showName.split(' ').forEach(function(str) {
      fetchString += str + '+'
    });
    fetchString += '&season=' + season + '&apikey=92e3aa84'
    fetch(fetchString)
      .then((response) => response.json())
      .then((responseJson) => {
        this.setState({
          episodes: responseJson.Episodes,
        });
      })
      .catch((error) => {
        console.error(error);
      });
  }

  render() {
    const { episodes } = this.state;
    const { navigation } = this.props;
    const showName = navigation.state.params.showName;
    const season = navigation.state.params.season;
    return (
      <View style={styles.container}>
        <Text style={[styles.Title, styles.FontColor]}>{showName}: Season {season} </Text>
        <FlatList
          data={episodes}
          renderItem={({ item }) => (
            <TouchableOpacity style={styles.BulletItem}
                              onPress={() => navigation.navigate('Episode', { episode: item.Episode, season: season, showName: showName})}>
              <Text style={styles.FontColor}>Episode: {item.Episode}: {item.Title} </Text>
            </TouchableOpacity>
          )}
        />
      </View>
    );
  }
}
