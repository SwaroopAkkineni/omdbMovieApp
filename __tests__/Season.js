import React from 'react';
import 'react-native';
import Season from '../src/Season';

import renderer from 'react-test-renderer';
const navigationMock = { state: { params: { season: "TestSeason", showName: "TestShow" } } };

const mockResponse = (status, statusText, response) => {
        return new Response(response, {
            status: status,
            statusText: statusText,
            headers: {
                'Content-type': 'application/json'
            }
        });
    };

test('renders correctly', () => {
  fetch = jest.fn(() => new Promise(resolve => resolve()));
  const tree = renderer.create(<Season navigation={navigationMock}/>).toJSON();
  expect(tree).toMatchSnapshot();
});
