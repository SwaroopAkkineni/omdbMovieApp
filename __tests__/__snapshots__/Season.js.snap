// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<View
  style={
    Object {
      "backgroundColor": "#89D2DC",
      "flex": 1,
      "paddingLeft": 75,
      "paddingRight": 75,
      "textAlign": "left",
    }
  }
>
  <Text
    style={
      Array [
        Object {
          "fontSize": 40.019999999999996,
          "fontWeight": "bold",
          "marginBottom": 37.5,
          "marginTop": 37.5,
        },
        Object {
          "color": "#101D42",
        },
      ]
    }
  >
    TestShow
    : Season 
    TestSeason
     
  </Text>
  <RCTScrollView
    data={Array []}
    disableVirtualization={false}
    getItem={[Function]}
    getItemCount={[Function]}
    horizontal={false}
    initialNumToRender={10}
    keyExtractor={[Function]}
    maxToRenderPerBatch={10}
    numColumns={1}
    onContentSizeChange={[Function]}
    onEndReachedThreshold={2}
    onLayout={[Function]}
    onMomentumScrollEnd={[Function]}
    onScroll={[Function]}
    onScrollBeginDrag={[Function]}
    onScrollEndDrag={[Function]}
    removeClippedSubviews={false}
    renderItem={[Function]}
    scrollEventThrottle={50}
    stickyHeaderIndices={Array []}
    updateCellsBatchingPeriod={50}
    viewabilityConfigCallbackPairs={Array []}
    windowSize={21}
  >
    <View />
  </RCTScrollView>
</View>
`;
